src/commonMain/kotlin/org/kosat/heuristics/Selector.kt

```diff
@@ -0,0 +1,63 @@
+package org.kosat.heuristics
+
+import org.kosat.CDCL
+import org.kosat.Clause
+import org.kosat.Lit
+import kotlin.math.abs
+
+abstract class Selector {
+    protected var assumptions: List<Lit> = emptyList()
+
+    fun initAssumptions(assumptions: List<Lit>) {
+        this.assumptions = assumptions
+    }
+
+    abstract fun build(clauses: List<Clause>)
+    abstract fun nextDecisionVariable(vars: List<CDCL.VarState>, level: Int): Int
```

Comment:
If you are going to rename "Selector" to "DecisionVariableSelector", this method could also be renamed to "next/select" in order to reduce the tautology (in the outer code, in the expression like `this.decisionVariableSelector.nextDecisionVariable()`, which is too mouthful).